packages/sitecore-jss-nextjs/src/services/graphql-sitemap-service.ts

```diff
@@ -61,225 +37,116 @@ export interface GraphQLSitemapServiceConfig {
+   * @returns an array of @see StaticPath objects
    */
-  async fetchExportSitemap(locale: string, rootItemPath: string): Promise<StaticPath[]> {
+  async fetchExportSitemap(locale: string): Promise<StaticPath[]> {
     const formatPath = (path: string[]) => ({
       params: {
         path,
       },
     });
 
-    return this.fetchSitemap([locale], rootItemPath, formatPath);
+    return this.fetchSitemap([locale], formatPath);
   }
 
   /**
    * Fetch sitemap which could be used for generation of static pages using SSG mode
    * @param {string[]} locales locales which application supports
-   * @param {string} rootItemPath root item path
-   * @throws {RangeError} if a valid root item ID cannot be determined from the specified root item path.
-   * @throws {RangeError} if the specified locale is not valid.
+   * @returns an array of @see StaticPath objects
    */
-  async fetchSSGSitemap(locales: string[], rootItemPath: string): Promise<StaticPath[]> {
+  async fetchSSGSitemap(locales: string[]): Promise<StaticPath[]> {
     const formatPath = (path: string[], locale: string) => ({
       params: {
         path,
       },
       locale,
     });
 
-    return this.fetchSitemap(locales, rootItemPath, formatPath);
+    return this.fetchSitemap(locales, formatPath);
   }
 
   /**
-   * Gets the list of site pages.
-   * @param {string[]} languages
-   * @param {string} rootItemPath
-   * @param {Function} formatStaticPath
-   * @default Search query
-   * query SitePageQuery(
-   *   $rootItemId: String!,
-   *   $language: String!,
-   *   $pageSize: Int = 10,
-   *   $after: String
-   * ) {
-   *   search(
-   *     where: {
-   *       AND:[
-   *         { name: "_path",      value: $rootItemId },
-   *         { name: "_language",  value: $language   },
-   *         { name: "_hasLayout", value: "true"      }
-   *       ]
-   *     }
-   *     first: $pageSize
-   *     after: $after
-   *   ) {
-   *     total
-   *     pageInfo {
-   *       endCursor
-   *       hasNext
-   *     }
-   *     results {
-   *       url {
-   *         path
-   *       }
-   *     }
-   *   }
-   * }
-   * @throws {RangeError} if a valid root item ID cannot be determined from the specified root item path.
-   * @throws {RangeError} if the specified language list is not valid.
+   * Fetch a flat list of all pages that are descendants of the specified root item and have a
+   * version in the specified language(s).
+   * @param {string[]} languages Fetch pages that have versions in this language(s).
+   * @param {Function} formatStaticPath Function for transforming the raw search results into (@see StaticPath) types.
+   * @returns list of pages
+   * @throws {RangeError} if the list of languages is empty.
+   * @throws {Error} if the app root was not found for the specified site and language.
    */
   async fetchSitemap(
     languages: string[],
-    rootItemPath: string,
     formatStaticPath: (path: string[], language: string) => StaticPath
```

Comment:
Per our discussion, our decision is to add support for 'excludes' paths since this is a common use cases. For other customization, the query needs to be overwritten by extending the service class.